presto-client/src/main/java/com/facebook/presto/client/QueryError.java

```diff
@@ -113,4 +115,29 @@ public String toString()
                 .add("failureInfo", failureInfo)
                 .toString();
     }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        QueryError that = (QueryError) o;
+        return errorCode == that.errorCode &&
+                Objects.equals(message, that.message) &&
+                Objects.equals(sqlState, that.sqlState) &&
+                Objects.equals(errorName, that.errorName) &&
+                Objects.equals(errorType, that.errorType) &&
+                Objects.equals(errorLocation, that.errorLocation) &&
+                Objects.equals(failureInfo, that.failureInfo);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(message, sqlState, errorCode, errorName, errorType, errorLocation, failureInfo);
```

Comment:
please order the same as in equals so it's obvious whether all are here

(why is equals required?)